src/api/channel/channel.ts

```diff
@@ -0,0 +1,179 @@
+type ChannelId = string;
+
+/**
+ * Object representing a context channel.
+ * 
+ * All interactions with a context channel happen through the methods on here.
+ */
+declare class Channel {
+    /**
+     * Constant that uniquely identifies this channel. Will be generated by the service, and guarenteed to be unique
+     * within the set of channels registered with the service.
+     * 
+     * In the case of `desktop` channels (see {@link DesktopChannel}), these id's _should_ persist across sessions. The 
+     * channel list is defined by the service, but can be overridden by a desktop owner. If the desktop owner keeps 
+     * this list static (which is recommended), then id's will also persist across sessions.
+     */
+    public readonly id: ChannelId;
+
+    /**
+     * Uniquely defines each channel type.
+     * 
+     * See overrides of this class for list of allowed values.
+     */
+    public readonly type: string;
+
+    /**
+     * Broadcasts the given context on this channel. This is equivilant to joining the channel and then calling the 
+     * top-level FDC3 `broadcast` function.
+     * 
+     * Note that this function can be used without first joining the channel, allowing applications to broadcast on
+     * channels that they aren't a member of.
+     * 
+     * This broadcast will be received by all windows that are members of this channel, *except* for the window that
+     * makes the broadcast. This matches the behaviour of the top-level FDC3 `broadcast` function.
+     */
+    public broadcast(context: Context): Promise<void>;
+
+    /**
+     * Returns the last context that was broadcast on this channel. All channels initially have no context, until a 
+     * window is added to the channel and then broadcasts. If there is not yet any context on the channel, this method
+     * will return `null`. The context is also reset back into it's initial context-less state whenever a channel is
+     * cleared of all windows.
+     * 
+     * The context of a channel will be captured regardless of how the context is broadcasted on this channel - whether
+     * using the top-level FDC3 `broadcast` function, or using the channel-level {@link broadcast} function on this 
+     * object.
+     * 
+     * NOTE: Only non-default channels are stateful, for the default channel this method will always return `null`.
+     */
+    public getCurrentContext(): Promise<Context|null>;
+
+    /**
+     * Event that is fired whenever a window broadcasts on this channel.
+     * 
+     * The `channel` property within the event will always be this channel instance.
+     */
+    public addEventListener(event: 'broadcast', listener: (event: {channel: Channel; context: Context}) => void): void;
+
+    /**
+     * Remove this event listener
+     * 
+     */
+    public removeEventListener(event: 'broadcast', listener: (event: {channel: Channel; context: Context}) => void): void;
```

Comment:
Rather than remove by reference, I'd recommend adopting the existing pattern, where `addEventListener` returns a `Listener` object that allows you to unsubscribe:

https://github.com/FDC3/FDC3/blob/master/src/api/interface.ts#L171

Reply:
This is done